src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -413,10 +421,14 @@ public static void GetDayOfMonth_Invalid(Calendar calendar, DateTime dt)
         [MemberData(nameof(DateTime_TestData))]
         public static void GetDayOfWeek_Invalid(Calendar calendar, DateTime dt)
         {
-            // TODO: this fails (expected) for lunar and lunarsolar calendars.
-            // Calendar.AlgorithmType is not exposed in corefx so how do we test for this?
-            // HijiriCalendar, UmAlQuraCalendar, PersianCalendar, HebrewCalendar
-            // Assert.Throws<ArgumentOutOfRangeException>("time", () => calendar.GetDayOfWeek(dt));
+            if (calendar is HijriCalendar || calendar is UmAlQuraCalendar || calendar is PersianCalendar || calendar is HebrewCalendar)
```

Comment:
same as previous comment. I just want to make sure we don't have bugs with such calendars and also we can try to handle the different in the data instead in the code.